File: CS/19_OLE/ModifyOLEData.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Spire.Presentation;
using System.IO;

namespace ModifyOLEData
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();            
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            //Create a PPT document
            Presentation presentation = new Presentation();

            //Load document from disk
            presentation.LoadFromFile(@"..\..\..\..\..\..\Data\ModifyOLEData.pptx");

            //Loop through the slides and shapes
            foreach (ISlide slide in presentation.Slides)
            {
                foreach (IShape shape in slide.Shapes)
                {
                    if (shape is IOleObject)
                    {
                        //Find OLE object
                        IOleObject oleObject = shape as IOleObject;

                        //Get its data and write to file
                        byte[] bytes = oleObject.Data;
                        MemoryStream pptStream = new MemoryStream(bytes);
                        MemoryStream stream=new MemoryStream();
                        if (oleObject.ProgId == "PowerPoint.Show.12")
                        {
                            //Load the PPT stream
                            Presentation ppt = new Presentation();
                            ppt.LoadFromStream(pptStream, Spire.Presentation.FileFormat.Auto);
                            //Append an image in slide
                            ppt.Slides[0].Shapes.AppendEmbedImage(ShapeType.Rectangle, @"..\..\..\..\..\..\Data\Logo.png", new RectangleF(50, 50, 100, 100));
                            ppt.SaveToFile(stream, Spire.Presentation.FileFormat.Pptx2013);
                            stream.Position = 0;
                            //Modify the data
                            oleObject.Data = stream.ToArray();
                        }
                    }
                }
            }

            //Save the document
            string result = "ModifyOLEData_result.pptx";
            presentation.SaveToFile(result, FileFormat.Pptx2013);

            //Launch the file
            OutputViewer(result);
        }
        private void OutputViewer(string filename)
        {
            try
            {
                System.Diagnostics.Process.Start(filename);
            }
            catch { }
        }
    }
}

File: VB/19_OLE/ModifyOLEData.vb
Imports System.ComponentModel
Imports System.Text
Imports Spire.Presentation
Imports System.IO

Namespace ModifyOLEData
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
			'Create a PPT document
			Dim presentation As New Presentation()

			'Load document from disk
			presentation.LoadFromFile("..\..\..\..\..\..\Data\ModifyOLEData.pptx")

			'Loop through the slides and shapes
			For Each slide As ISlide In presentation.Slides
				For Each shape As IShape In slide.Shapes
					If TypeOf shape Is IOleObject Then
						'Find OLE object
						Dim oleObject As IOleObject = TryCast(shape, IOleObject)

						'Get its data and write to file
						Dim bytes() As Byte = oleObject.Data
						Dim pptStream As New MemoryStream(bytes)
						Dim stream As New MemoryStream()
						If oleObject.ProgId = "PowerPoint.Show.12" Then
							'Load the PPT stream
							Dim ppt As New Presentation()
							ppt.LoadFromStream(pptStream, Spire.Presentation.FileFormat.Auto)
							'Append an image in slide
							ppt.Slides(0).Shapes.AppendEmbedImage(ShapeType.Rectangle, "..\..\..\..\..\..\Data\Logo.png", New RectangleF(50, 50, 100, 100))
							ppt.SaveToFile(stream, Spire.Presentation.FileFormat.Pptx2013)
							stream.Position = 0
							'Modify the data
							oleObject.Data = stream.ToArray()
						End If
					End If
				Next shape
			Next slide

			'Save the document
			Dim result As String = "ModifyOLEData_result.pptx"
			presentation.SaveToFile(result, FileFormat.Pptx2013)

			'Launch the file
			OutputViewer(result)
		End Sub
		Private Sub OutputViewer(ByVal filename As String)
			Try
				Process.Start(filename)
			Catch
			End Try
		End Sub
	End Class
End Namespace